pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -9115,22 +9117,21 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 
 		// Emit resolved events for majority of spans.  Be extra paranoid and ensure that
 		// we have at least 1 span for which we don't emit resolved timestamp (to force checkpointing).
-		haveGaps := false
+		// We however also need to ensure there's at least one span that isn't filtered out.
+		var allowedOne, haveGaps bool
 		knobs.FilterSpanWithMutation = func(r *jobspb.ResolvedSpan) (bool, error) {
 			if r.Span.Equal(tableSpan) {
-				// Do not emit resolved events for the entire table span.
```

Comment:
why remove this bit? seems potentially useful

Reply:
I had to remove the analogous piece of code in `TestAlterChangefeedAddTargetsDuringBackfill` in #153758 because it was causing rare flakes where this fake span got flushed to the change frontier and showed up in the checkpoint, which caused a test assertion to fail. Preemptively removing it here in case it causes similar flakes.